yaksh/models.py

```diff
@@ -1216,11 +1216,18 @@ def can_attempt_now(self, user, course_id):
             )
             if last_attempt:
                 time_lag = (timezone.now() - last_attempt.start_time).total_seconds() / 3600
-                return time_lag >= self.quiz.time_between_attempts
+                can_attempt = time_lag >= self.quiz.time_between_attempts
+                msg = "You cannot start the next attempt for this quiz before {0} hour(s)".format(
+                    self.quiz.time_between_attempts
+                ) if not can_attempt else None
+                return can_attempt, msg
             else:
-                return True
+                return True, None
```

Comment:
This is why doing it in views make sense. Sending None is pointless

Reply:
I don't think that it is pointless, ```None``` just means that there is no error message. Pushing this to the views will mean adding if statements in a place that is already crowded with a lot of decision making in the first place (lines 543 to line 559 or so).

Returning all the necessary data from the models keeps the views clean.

Reply:
Fair enough.

Reply:
I just believe that a method shouldn't return a message. To me, I don't believe model methods should do that.